packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -112,179 +103,119 @@ const propTypes = {
     return null;
   }
+
+  const actionButtons = [];
+
   if (acceptAction) {
-    acceptButton = (emphasizedAction === 'accept')
-      ? <Button text={acceptAction.text} variant={ButtonVariants.EMPHASIS} onClick={acceptAction.onClick} />
-      : <Button text={acceptAction.text} onClick={acceptAction.onClick} />;
-  }
-  if (rejectAction) {
-    rejectButton = (emphasizedAction === 'reject')
-      ? <Button text={rejectAction.text} variant={ButtonVariants.EMPHASIS} onClick={rejectAction.onClick} />
-      : <Button text={rejectAction.text} onClick={rejectAction.onClick} />;
+    const buttonVariant = emphasizedAction === 'accept' ? { variant: ButtonVariants.EMPHASIS } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="accept" key="accept" text={acceptAction.text} onClick={acceptAction.onClick} />);
   }
 
-  if (buttonOrder === 'rejectFirst') {
-    return (
-      <div className={cx('actions')}>
-        {rejectButton}
-        {acceptButton}
-      </div>
-    );
+  if (rejectAction) {
+    const buttonVariant = emphasizedAction === 'reject' ? { variant: ButtonVariants.EMPHASIS } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="reject" key="reject" text={rejectAction.text} onClick={rejectAction.onClick} />);
   }
 
   return (
     <div className={cx('actions')}>
-      {acceptButton}
-      {rejectButton}
+      {buttonOrder === 'acceptFirst' ? actionButtons : actionButtons.reverse()}
     </div>
   );
 };
 
-const getIcon = (variant, customIcon = null) => {
-  switch (variant) {
-    case variants.ALERT:
-      return (<span className={cx(['icon', 'alert'])} />);
-    case variants.ERROR:
-      return (<span className={cx(['icon', 'error'])} />);
-    case variants.WARNING:
-      return (<span className={cx(['icon', 'warning'])} />);
-    case variants.INFO:
-      return (<span className={cx(['icon', 'info'])} />);
-    case variants.SUCCESS:
-      return (<span className={cx(['icon', 'success'])} />);
-    case variants.CUSTOM:
-      return customIcon;
-    default:
-      return null;
+const NotificationDialog = (props) => {
+  const theme = React.useContext(ThemeContext);
+
+  const {
+    title,
+    startMessage,
+    endMessage,
+    content,
+    acceptAction,
+    rejectAction,
+    variant,
+    isOpen,
+    buttonOrder,
+    emphasizedAction,
+    custom,
+    ...customProps
+  } = props;
+
+  if (isOpen && acceptAction === undefined && rejectAction === undefined) {
+    throw new Error('Unable to render the Notification dialog. Neither an `acceptAction` and/or `rejectAction` prop was provided which leaves the user without the ability to close the dialog.');
   }
-};
-
-class NotificationDialog extends React.Component {
-  constructor(props) {
-    super(props);
 
-    this.handleKeydown = this.handleKeydown.bind(this);
-    this.escapeKey = KeyCode.KEY_ESCAPE;
+  if (isOpen && !isValidVariant(variant)) {
```

Comment:
Is parsing the list of variants necessary for this logic path or can it be simplified to checking if `variant` was provided?

```jsx
if (isOpen && !variant) {
```

I think we should leave the validation checking to prop-types

Reply:
If we throw this error when isOpen = false, the whole page crashes, rather than when the dialog is trying to render.

Reply:
Can you follow-up to this comment? Was it indented to be a response to this particular thread?

Reply:
removed backed on feedback on this thread: https://github.com/cerner/terra-framework/pull/1105#discussion_r424676883

https://github.com/cerner/terra-framework/pull/1105/commits/3ee7c3e4fcb737ea8b167a829b5c9b2ea7d1935e